api-docs: Update mute and unmute user endpoint documentation.

laurynmm opened the pull request:
Updates the main descriptions for the [mute a user](https://zulip.com/api/mute-user) and [unmute a user](https://zulip.com/api/unmute-user) endpoint documentation. Also, revises the `muted_user_id` parameter description and **Changes** note for feature level 188.

The feature level 188 changes were made in #26005.

**Screenshots and screen captures:**

<details>
<summary>Mute a user - main description</summary>

[Current documentation](https://zulip.com/api/mute-user)
![Screenshot from 2023-06-14 13-31-52](https://github.com/zulip/zulip/assets/63245456/2aec7e98-5ac4-4482-8a78-ec84c379f806)
</details>
<details>
<summary>Unmute a user - main description</summary>

[Current documentation](https://zulip.com/api/unmute-user)
![Screenshot from 2023-06-14 13-32-20](https://github.com/zulip/zulip/assets/63245456/fe13fdb7-1045-4de8-a535-7e0403ebf501)
</details>
<details>
<summary>muted_user_id parameter</summary>

[Current documentation](https://zulip.com/api/mute-user#parameter-muted_user_id)
![Screenshot from 2023-06-14 13-32-35](https://github.com/zulip/zulip/assets/63245456/5956ce0f-1a40-4366-8ee2-aaec93f53a46)
</details>

---

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

gnprice commented on zerver/openapi/zulip.yaml:
```diff
@@ -8961,8 +8961,8 @@ paths:
       description: |
-        This endpoint [mutes a user](/help/mute-a-user). Messages sent by users
-        you've muted will be automatically marked as read and hidden.
+        [Mute a user](/help/mute-a-user). Messages sent by muted users
+        will be automatically marked as read and hidden.
```
This wording sounds a lot like it's about setting a flag on the muted user globally. It'd be good to add a few words to clarify that it's only about a relationship between the user being muted and the user who's doing the muting.

Perhaps:
```suggestion
        [Mute a user](/help/mute-a-user) as seen by the requesting user.
        Messages sent by muted users will be automatically marked as read and hidden
        from the perspective of the user who muted them.
```

gnprice replied:
(The existing text isn't super clear about this either, but I think the "*you've* muted" helps)

laurynmm replied:
Also added the "from the perspective of the requesting user" to the `api/unmute-user` endpoint description.

laurynmm commented:
**Updates**:
- Updated main description for both endpoints for review comment above. See updated screenshots below.
- Also, made two other small tweaks to the `api/mute-user` endpoint description:
  - Replaced the other "you" reference to be the "requesting user"
  - Changed `"1:1"` to be `"one-on-one"`. This was the only place in the API documentation that we used `"1:1"` for direct messages. Generally for user-facing documentation, I think it would be better to use `"one-on-one"` for referring to these direct messages as it will likely be translated by the browser / third parties more correctly than the abbreviated notation.

**Updated screenshots**:

<details>
<summary>Mute a user - main description</summary>

[Current documentation](https://zulip.com/api/mute-user)
![Screenshot from 2023-06-26 18-30-48](https://github.com/zulip/zulip/assets/63245456/949636ce-05f4-43cd-99cc-0f13694fbdd0)
</details>
<details>
<summary>Unmute a user - main description</summary>

[Current documentation](https://zulip.com/api/unmute-user)
![Screenshot from 2023-06-26 18-30-55](https://github.com/zulip/zulip/assets/63245456/b78ca685-4b3c-4e64-9bf8-b4da3db514ce)
</details>

gnprice commented:
LGTM!

timabbott commented:
Merged, thanks @laurynmm and @gnprice!
